data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfiguration.java

```diff
@@ -180,8 +180,14 @@ private void attachAWSSigV4Callback(final RestClientBuilder restClientBuilder) {
     final AWSCredentialsProvider credentialsProvider = new DefaultAWSCredentialsProviderChain();
     final HttpRequestInterceptor httpRequestInterceptor = new AWSRequestSigningApacheInterceptor(SERVICE_NAME, aws4Signer,
             credentialsProvider);
-    restClientBuilder.setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.addInterceptorLast(
-            httpRequestInterceptor));
+    final SSLContext sslContext = certPath != null ? getCAStrategy(certPath) : getTrustAllStrategy();
+    restClientBuilder.setHttpClientConfigCallback(httpClientBuilder -> {
+      httpClientBuilder.setSSLContext(sslContext).addInterceptorLast(httpRequestInterceptor);
+      if (this.insecure) {
+        httpClientBuilder.setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE);
+      }
+      return httpClientBuilder;
+    });
```

Comment:
This code is repeated in both AWSSigV4 and username methods, can you not attach. the ssl in the `createClient` instead?

Reply:
Due to the callback structure, I refactored out `attachSSLContext(final HttpAsyncClientBuilder httpClientBuilder)` and invoked it in both.